bootstrap/config/config.go

```diff
@@ -550,3 +556,34 @@ func (cp *Processor) listenForChanges(serviceConfig interfaces.Configuration, co
 func (cp *Processor) logConfigInfo(message string, overrideCount int) {
 	cp.lc.Infof("%s (%d envVars overrides applied)", message, overrideCount)
 }
+
+// getSecretPathsChanged returns an array of changed secrets and paths between two maps.
+func getSecretPathsChanged(m1 config.InsecureSecrets, m2 config.InsecureSecrets) []string {
+	var updatedPaths []string
+	for key1, val1 := range m1 {
+		if val1.Path == m2[key1].Path {
+			for k, v := range val1.Secrets {
+				// Checking for changed secret.
+				if m2[key1].Secrets[k] != v {
+					updatedPaths = append(updatedPaths, val1.Path)
+				}
+			}
+		} else {
+			// Handling for changes to a Path.
+			updatedPaths = append(updatedPaths, val1.Path, m2[key1].Path)
+		}
+	}
+
+	return cleanSecretPathArray(updatedPaths)
+}
+
+// cleanSecretPathArray cleans an array of secret paths.
```

Comment:
This comment is not descriptive enough as to why this is needed.  Why put a blank path in the map to begin with.  Could we have not checked this earlier instead of doing another pass on the array?